internal/controller/prometheus_manager.go

```diff
@@ -92,13 +82,12 @@ func (r *InstasliceReconciler) UpdateCompatibleProfilesMetrics(instasliceObj inf
 			for _, gpuID := range sortedGPUs {
 				fit, err := r.calculateProfileFitOnGPU(&instasliceObj, profileName, gpuID, true, nil)
 				if err != nil {
-					return fmt.Errorf("failed to calculate compatible profiles for profile %s: %w", profileName, err)
```

Comment:
Good to keep this error and just log the same in the main loop without disrupting the main sync.